server/src/com/cloud/network/security/SecurityGroupManagerImpl.java

```diff
@@ -437,10 +434,6 @@ public void doInTransactionWithoutResult(TransactionStatus status) {
                     }
                 }
             });
-
-            for (Long vmId : affectedVms) {
```

Comment:
@rafaelweingartner why have you removed this?

Reply:
Because it was not making much sense to me; for every VM ID in `affectedVms` it executes an asynchronous task  `_executorPool.schedule(new WorkerThread(), delayMs, TimeUnit.MILLISECONDS);`. The odd part is that the `vmId` is not used. All of the tasks are scheduled to be executed are the same.

Looking at com.cloud.network.security.SecurityGroupManagerImpl.work() closely, it seems that it is indeed used for something. I was expecting it to be a generic task where we send something to be computed on. 

So, there this idea of security group work, and the guys that implemented this, probably wanted to execute/process a number of work that is equal to the number of `affectedVms`. It would have been clear if done in some other way… I will not refactor this.
I will re-added this bit again.

@rhtyd  thanks for spotting that!